manage-data/data-store/data-streams/failure-store-recipes.md

```diff
@@ -0,0 +1,1153 @@
+          "match": { <2>
+            "document.index": "my-datastream-ingest-example"
+          }
+        },
+        {
+          "match": { <3>
+            "error.type": "illegal_argument_exception"
+          }
+        },
+        {
+          "range": { <4>
+            "@timestamp": {
+              "gt": "2025-05-01T00:00:00Z",
+              "lte": "2025-05-02T00:00:00Z"
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+1. Require the `error.pipeline` field to exist. This filters to ingest pipeline failures only.
+2. Filter on the data stream name to remediate documents headed for a specific index.
+3. Further narrow which kind of failure you are attempting to remediate. In this example we are targeting a specific type of error.
+4. Filter on timestamp to only retrieve failures before a certain point in time. This provides a stable set of documents.
+
+Take note of the documents that are returned. We can use these to simulate that our remediation logic makes sense
+```console-result
+{
+  "took": 14,
+  "timed_out": false,
+  "_shards": {
+    "total": 2,
+    "successful": 2,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 2.575364,
+    "hits": [
+      {
+        "_index": ".fs-my-datastream-ingest-example-2025.05.16-000001",
+        "_id": "cOnR2ZYByIwDXH-g6GpR",
+        "_score": 2.575364,
+        "_source": {
+          "@timestamp": "2025-05-01T15:58:53.522Z", <1>
+          "document": {
+            "index": "my-datastream-ingest-example",
+            "source": {
+              "@timestamp": "2025-05-01T00:00:00Z",
+              "data": {
+                "counter": 42 <2>
+              }
+            }
+          },
+          "error": {
+            "type": "illegal_argument_exception",
+            "message": "field [id] not present as part of path [data.id]", <3>
+            "stack_trace": """j.l.IllegalArgumentException: field [id] not present as part of path [data.id]
+	at o.e.i.IngestDocument.getFieldValue(IngestDocument.java:202)
+	at o.e.i.c.SetProcessor.execute(SetProcessor.java:86)
+	... 14 more
+""",
+            "pipeline_trace": [
+              "my-datastream-default-pipeline"
+            ],
+            "pipeline": "my-datastream-default-pipeline", <4>
+            "processor_type": "set"
+          }
+        }
+      }
+    ]
+  }
+}
+```
+1. This document is what we'll use for our simulations.
+2. It had a counter value.
+3. The document was missing a required field.
+4. The document failed in the `my-data-stream-default-pipeline`
+   ::::
+
+::::{step} Fix the original problem
+Because ingest pipeline failures need to be reprocessed by their original pipelines, any problems with those pipeline should be fixed before remediating failures. Investigating the pipeline mentioned in the example above shows that there is a processor that expects a field to be present that is not always present.
```

Comment:
```suggestion
Because ingest pipeline failures need to be reprocessed by their original pipelines, any problems with those pipelines should be fixed before remediating failures. Investigating the pipeline mentioned in the example above shows that there is a processor that expects a field to be present that is not always present.
```